pkg/operator/sync.go

```diff
@@ -635,6 +637,16 @@ func getProxyArgs(config *OperatorConfig) []corev1.EnvVar {
 	return envVars
 }
 
+// buildFeatureGatesString builds a string with the format: --feature-gates=<name>=<bool>,<name>=<bool>...
+func buildFeatureGatesString(featureGates map[string]bool) string {
+	var parts []string
+	for name, enabled := range featureGates {
+		part := fmt.Sprintf("%s=%t", name, enabled)
+		parts = append(parts, part)
+	}
+	return "--feature-gates=" + strings.Join(parts, ",")
```

Comment:
When we merge this, will it then break all of the operands? If they are erroring on unknown flags we will have to add the flag to the operand first (doing nothing?) and then merge this

Reply:
I think this is the case (e.g looking at machine-api-provider-aws) will fail with an error like: `flag provided but not defined: -feature-gates`